sources/osgShadow/ShadowMap.js

```diff
@@ -236,7 +236,7 @@ define( [
 
         // Overridable Visitor so that user can override the visitor to enable disable
         // in its own shadowmap implementation
-        this._removeNodesNeverCastingVisitor = new RemoveNodesNeverCastingVisitor( this._castsShadowTraversalMask );
+        this._removeNodesNeverCastingVisitor = settings.userCallbackNodeCaster || new RemoveNodesNeverCastingVisitor( this._castsShadowTraversalMask );
```

Comment:
I am not sure about passing function or class instance in settings, it's not used for all other osgjs code.